bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/handler/QolsysIQPartitionHandler.java

```diff
@@ -0,0 +1,357 @@
+import org.openhab.binding.qolsysiq.internal.client.dto.event.ErrorEvent;
+import org.openhab.binding.qolsysiq.internal.client.dto.event.SecureArmInfoEvent;
+import org.openhab.binding.qolsysiq.internal.client.dto.event.ZoneActiveEvent;
+import org.openhab.binding.qolsysiq.internal.client.dto.event.ZoneAddEvent;
+import org.openhab.binding.qolsysiq.internal.client.dto.event.ZoneUpdateEvent;
+import org.openhab.binding.qolsysiq.internal.client.dto.model.AlarmType;
+import org.openhab.binding.qolsysiq.internal.client.dto.model.Partition;
+import org.openhab.binding.qolsysiq.internal.client.dto.model.PartitionStatus;
+import org.openhab.binding.qolsysiq.internal.client.dto.model.Zone;
+import org.openhab.binding.qolsysiq.internal.config.QolsysIQPartitionConfiguration;
+import org.openhab.binding.qolsysiq.internal.discovery.QolsysIQChildDiscoveryService;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link QolsysIQPartitionHandler} manages security partitions
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class QolsysIQPartitionHandler extends BaseBridgeHandler implements QolsysIQChildDiscoveryHandler {
+    private final Logger logger = LoggerFactory.getLogger(QolsysIQPartitionHandler.class);
+    private static final int CLEAR_ERROR_MESSSAGE_TIME = 30;
+    private @Nullable QolsysIQChildDiscoveryService discoveryService;
+    private @Nullable ScheduledFuture<?> delayFuture;
+    private @Nullable ScheduledFuture<?> errorFuture;
+    private @Nullable Partition partitionCache;
+    private @Nullable String armCode;
+    private @Nullable String disarmCode;
+    private List<Zone> zones = Collections.synchronizedList(new LinkedList<Zone>());
+    private int partitionId;
+
+    public QolsysIQPartitionHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        QolsysIQPartitionConfiguration config = getConfigAs(QolsysIQPartitionConfiguration.class);
+        partitionId = config.id;
+        armCode = config.armCode.strip().length() == 0 ? null : config.armCode;
+        disarmCode = config.disarmCode.strip().length() == 0 ? null : config.disarmCode;
+        logger.debug("initialize partition {}", partitionId);
+        refresh();
```

Comment:
There are specific cases that will trigger no set of status, for example if no bridge is defined for that thing.
I would suggest something like that:
```
        QolsysIQPanelHandler panel = panelHandler();
        if (panel == null) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
        } else if (panel.getThingStatus() != ThingStatus.ONLINE) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
        } else {
            updateStatus(ThingStatus.UNKNOWN);
            scheduler.execute(() -> {
                panel.refresh();
            });
        }
```

Reply:
Ah, yes thats a good call.